### PR TITLE
SSTable set improvements

### PR DESCRIPTION
**Pull request by raphaelsc**
Makes sstable_set::all() interface robust, and introduces sstable_set::size() to avoid copies when retrieving set size.

### CODE REVIEW

File: sstables/sstable_set.cc [hunk 1]
```diff
@@ -175,6 +175,11 @@ sstable_set::erase(shared_sstable sst) {
     _impl->erase(sst);
 }
 
+size_t
+sstable_set::size() const {
```
Comment (bhalevy) [inline sstables/sstable_set.cc:179]: please define as noexcept so it can be safely used in exception sensitive paths.

**Review by bhalevy — APPROVED** [review]
Looks good to me.
Please just define size as noexcept

**Comment by raphaelsc** [comment]
> Looks good to me. Please just define size as noexcept

After I sent it, I thought: Benny will probably ask me to noexceptify the interface ;-). Will do it in v2. thanks.

**Comment by raphaelsc** [comment]
v2: noexceptify sstable_set:size().

**Review by bhalevy — APPROVED** [review]

**Review by denesb — APPROVED** [review]

**Pull request merged**